### PR TITLE
[0.34] Custom resource management

### PR DESCRIPTION
**Pull request by jwitz**

### CODE REVIEW

File: software/customize-resource-usage.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+sidebar_label: 'Customize resource usage'
+title: 'Customize Deployment CPU and management resources per component'
+id: customize-resource-usage
+description: Scale Deployments directly using non-proportional CPU and memory specifications.
+---
+
+By default, all Astronomer Deployments use Astronomer Units (AU) to define a relative amount of CPU and memory that you can allocate to a Deployment, and components must use the same AU value for both CPU and memory.
```
Comment (rishkarajgi) [inline software/customize-resource-usage.md:8]: This will be default for existing deployments. For new deployments the default resource strategy will be Custom Resources ie astroUnitsEnabled: false (flag for that deployment only, not global flag)

File: software/customize-resource-usage.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+You can change the amount of CPU and memory an AU represents, but there are some scenarios where you need to size Deployments differently and without restraints on how your CPU and memory scale together. For example, you might need to allocate significantly more memory than CPU to your worker Pods if you need to run memory-intensive tasks, but at the same time you need more memory than CPU in your scheduler. In this scenario, using AUs isn't sufficient because each component needs a different CPU to memory ratio.
+
+For the greatest degree of flexibility, you can enable custom resource specifications for Deployments. When this feature is enabled, Deployment Admins can specify the exact amount of CPU and memory they want each component to have without any scaling limitations. The resources you specify are used both as the limit and request values for the Pods running your components.
```
Comment (rishkarajgi) [inline software/customize-resource-usage.md:12]: `When this feature is enabled` -> This is always enabled, we can rephrase this as `with Custom Resource strategy`
Basically if astroUnitsEnabled : true it means AU strategy + Custom Resource strategy
If disabled it means only Custom Resource strategy

File: software/customize-resource-usage.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+## Set custom resource usage in the Software UI
+
+To switch from using AU to using custom resource specifications: 
+
+1. In the Software UI, open your Deployment.
+2. In the **Settings** tab, change your **Resource Strategy** to **Custom Resources**.
+3. For each Software component, adjust the CPU and memory sliders to match your resource requirements. You can change the unit of the resource by clicking the dropdown menu that shows your current unit.
+
```
Comment (rishkarajgi) [inline software/customize-resource-usage.md:21]: We can add a section for GraphQL APIs, that they will have to pass this flag to set custom Resources

1. Create Deployment 
```
{
  "astroUnitsEnabled": false,
}
```
2. Update Deployment
```
{
  "payload": {
    "astroUnitsEnabled": false,
  },
} 
```
4. Upsert Deployment
```
{
  "astroUnitsEnabled": false,
}
```

Comment (jwitz) [reply software/customize-resource-usage.md:21]: Added an info note for this below

**Review by rishkarajgi — APPROVED** [review]
lgtm
